src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts

```diff
@@ -0,0 +1,206 @@
+import { ConfigService, ResourceService, Framework, ToasterService, ServerResponse, } from '@sunbird/shared';
+import { Component, OnInit, Input, Output, EventEmitter, ApplicationRef, ChangeDetectorRef } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { FrameworkService, FormService } from './../../services';
+import * as _ from 'lodash';
+import { CacheService } from 'ng2-cache-service';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+  selector: 'app-data-driven-filter',
+  templateUrl: './data-driven-filter.component.html',
+  styleUrls: ['./data-driven-filter.component.css']
+})
+export class DataDrivenFilterComponent implements OnInit {
+  @Input() routerVal: string;
+  @Input() inPageFilter: boolean;
+  @Output() triggerParentSearch: EventEmitter<any> = new EventEmitter();
+  /**
+ * To get url, app configs
+ */
+  public config: ConfigService;
+
+  public resourceService: ResourceService;
+
+  public filterType: string;
+  /**
+ * To navigate to other pages
+ */
+  public router: Router;
+  /**
+* To show toaster(error, success etc) after any API calls
+*/
+  public toasterService: ToasterService;
+
+  public frameworkService: FrameworkService;
+
+  public formService: FormService;
+
+  public formFieldProperties: any;
+
+  public categoryMasterList: any;
+
+  public framework: string;
+
+  public exists: boolean;
+
+  public formType = 'content';
+
+  public formAction = 'search';
+
+  public pageNumber: Number = 1;
+
+  public queryParams: any;
+
+  public redirectUrl: any;
+  /**
+ * formInputData is to take input data's from form
+ */
+  public formInputData: any;
+
+  refresh = true;
+  isShowFilterPlaceholder = true;
+  /**
+    * Constructor to create injected service(s) object
+    Default method of Draft Component class
+    * @param {Router} route Reference of Router
+    * @param {PaginationService} paginationService Reference of PaginationService
+    * @param {ConfigService} config Reference of ConfigService
+  */
+  constructor(config: ConfigService,
+    resourceService: ResourceService,
+    router: Router,
+    private activatedRoute: ActivatedRoute,
+    private _cacheService: CacheService,
+    private cdr: ChangeDetectorRef,
+    frameworkService: FrameworkService,
+    formService: FormService,
+    toasterService: ToasterService,
+
+  ) {
+    this.config = config;
+    this.resourceService = resourceService;
+    this.router = router;
+    this.frameworkService = frameworkService;
+    this.formService = formService;
+    this.toasterService = toasterService;
+    this.formInputData = {};
+  }
+
+  ngOnInit() {
+    this.frameworkService.initialize();
+    this.formInputData = {};
+    this.getQueryParams();
+    this.filterType = this.routerVal;
```

Comment:
not need. routerVal is input param, what is the point in assigning value from one member to other